File: support_tools/swigwin-3.0.12/Examples/test-suite/ruby/integers_runme.rb
#!/usr/bin/env ruby
#
# Put script description here.
#

require 'swig_assert'
require 'integers'
include Integers

swig_assert_each_line <<EOF
signed_char_identity(-3)   == -3
unsigned_char_identity(5)  == 5
signed_short_identity(-3)  == -3
unsigned_short_identity(5) == 5
signed_int_identity(-3)  == -3
unsigned_int_identity(5) == 5
signed_long_identity(-3)  == -3
unsigned_long_identity(5) == 5
signed_long_long_identity(-3)  == -3
unsigned_long_long_identity(5) == 5
EOF

File: support_tools/swigwin-3.0.12/Examples/test-suite/r/integers_runme.R
clargs <- commandArgs(trailing=TRUE)
source(file.path(clargs[1], "unittest.R"))

dyn.load(paste("integers", .Platform$dynlib.ext, sep=""))
source("integers.R")
cacheMetaData(1)

unittest(signed_char_identity(1), 1)
unittest(unsigned_char_identity(1), 1)
unittest(signed_short_identity(1), 1)
unittest(unsigned_short_identity(1), 1)
unittest(signed_int_identity(1), 1)
unittest(unsigned_int_identity(1), 1)
unittest(signed_long_identity(1), 1)
unittest(unsigned_long_identity(1), 1)
unittest(signed_long_long_identity(1), 1)
unittest(unsigned_long_long_identity(1), 1)

q(save="no")



